### PR TITLE
Uniform Quick Action tile height

### PR DESCRIPTION
**Pull request by jphechter**
## Description

Slight refactor of Quick Actions to make component more consistent and tile heights uniform.

### Ticket

[#186810093](https://www.pivotaltracker.com/story/show/186810093)

## Code author checklist

- [ ] I have performed a self-review of my code
- [ ] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [ ] I have not used any relative imports
- [ ] I have checked for and removed instances of unused config from CMS
- [ ] I have pruned any instances of unused code
- [ ] I have not added any markdown to labels, titles and button text in config
- [ ] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [ ] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts`

### CODE REVIEW

File: web/src/components/dashboard/QuickActionTile.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  type: "link";
+  quickAction: QuickActionLink;
+}
+
+interface TaskProps {
+  type: "task";
+  quickAction: QuickActionTask;
+}
+
+export const QuickActionTile = (props: LinkProps | TaskProps): ReactElement => {
```
Comment (jphechter) [inline web/src/components/dashboard/QuickActionTile.tsx:17]: One of the topics we've covered in our tech video sessions is "making impossible types impossible". This is a [discriminated union](https://basarat.gitbook.io/typescript/type-system/discriminated-unions). Leveraging this lets us consolidate 2 separate use cases of this component without having to handle for extensive conditional logic, or as existed previously, 2 separate but similar component definitions.

In this specific instance you could argue that "link" and "task" could be combined further given that `urlSlug` and `externalRoute` are similar enough data, but I'm intentionally opting to keep that separate.

File: web/src/styles/components/quick-action-tile.scss [hunk 2]
```diff
@@ -19,6 +19,9 @@
 .quick-action-tile {
   background-color: $base-extra-light;
   border: 2px solid $base-lightest;
+  cursor: pointer;
+  height: 100%;
+  line-height: 160%;
```
Comment (seidior) [inline web/src/styles/components/quick-action-tile.scss:24]: [dust] Should we also disable highlighting too?

**Review by seidior — APPROVED** [review]
Sets good standards in our code base and reduces complexity.

File: web/src/styles/components/quick-action-tile.scss (hunk 2 above)
Comment (jphechter) [reply web/src/styles/components/quick-action-tile.scss:24]: Great idea. Included.

File: web/src/components/dashboard/QuickActionTile.test.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+          type="link"
+          quickAction={generateQuickActionLink({
+            name: "some name",
+            externalRoute: "some-url",
+            filename: "some-filename",
+          })}
+        />
+      );
+      fireEvent.click(screen.getByText("some name"));
+      expect(mockAnalytics.event.quick_action_button.click.go_to_quick_action_screen).toHaveBeenCalledWith(
```
Comment (FarazA22) [inline web/src/components/dashboard/QuickActionTile.test.tsx:45]: dust - should there be an assertion validating the routing doesn't fire?

Comment (jphechter) [reply web/src/components/dashboard/QuickActionTile.test.tsx:52]: included

**Pull request merged**